handlers/photo-haptics-audio-handler/src/server.ts

```diff
@@ -0,0 +1,327 @@
+
+            const centroidArray = [];
+            const coordArray = [];
+             for (let i = 0; i < objsByGroup.length; i++) {
+               const centroid = objsByGroup[i]["centroid"]
+               const coords = objsByGroup[i]["dimensions"]
+               centroidArray.push(centroid)
+               coordArray.push(coords)
+             }
+             groupCentroidArray.push(centroidArray);
+             groupCoordArray.push(coordArray);
+            }
+        for (const idx of preGroupData["ungrouped"]) {
+            const obj = preObjDet["objects"].find((x: { "ID": number }) => x["ID"] === idx);
+            const centroid = obj["centroid"]
+            const coords = obj["dimensions"]         
+            groupCentroidArray.push(centroid);
+            groupCoordArray.push(coords);
+        }
+    } else {
+        console.warn("No objects were detected.");
+    }
+
+    const image = req.body.image;
+
+    if (hasAudioHaptic) {
+        try {
+            // Do TTS
+            const ttsResponse = await utils.getTTS(ttsData.map(x => x["value"]));
+            // Add offset values to data
+            for (let i = 0, offset = 0; i < ttsData.length; i++) {
+                ttsData[i]["audio"] = {
+                    "offset": offset,
+                    "duration": ttsResponse.durations[i]
+                };
+                offset += ttsResponse.durations[i];
+            }
+
+            const scData = {
+                "data": ttsData,
+                "ttsFileName": ""
+            };
+
+            // Write to file
+            let inFile: string, outFile: string, jsonFile: string;
+            await fetch(ttsResponse["audio"]).then(resp => {
+                return resp.arrayBuffer();
+            }).then(async (buf) => {
+                inFile = filePrefix + req.body["request_uuid"] + ".wav";
+                await fs.writeFile(inFile, Buffer.from(buf));
+                scData["ttsFileName"] = inFile;
+                jsonFile = filePrefix + req.body["request_uuid"] + ".json";
+                await fs.writeFile(jsonFile, JSON.stringify(scData));
+                outFile = filePrefix + uuidv4() + ".flac";
+                await fs.writeFile(outFile, "");
+                await fs.chmod(outFile, 0o664);
+
+                console.log("Forming OSC...");
+                return utils.sendOSC(jsonFile, outFile, "supercollider", scPort);
+            }).then(async (segArray:any) => {
+                const buffer = await fs.readFile(outFile);
+                // TODO detect mime type from file
+                const dataURL = "data:audio/flac;base64," + buffer.toString("base64");
+                if (hasAudioHaptic && segArray.length > 0 
+                    && segArray.length > hapticSegInfo.length) {
+                        
+                    const s = [...segArray];
+
+                    // Empty for text fields

```

Comment:
I can't tell what's going on in this section...could you add some more comments/use more descriptive variable names?